src/elements/toniq-dropdown/toniq-dropdown.element.ts

```diff
@@ -0,0 +1,161 @@
+import {assign, css, defineElementEvent, html} from 'element-vir';
+import {ChevronDown24Icon} from '../../icons';
+import {interactionDuration, noUserSelect, toniqFontStyles} from '../../styles';
+import {applyBackgroundAndForeground, toniqColors} from '../../styles/colors';
+import {removeNativeButtonStyles} from '../../styles/native-styles';
+import {defineToniqElement} from '../define-toniq-element';
+import {ToniqIcon} from '../toniq-icon/toniq-icon.element';
+
+declare global {
+    interface Window {
+        dropdownListenerAdded: boolean;
+    }
+}
+window.dropdownListenerAdded = window.dropdownListenerAdded || false;
+
+export interface ToniqDropdownOption {
+    value: unknown;
+    label: string;
+}
+
+export const ToniqDropdown = defineToniqElement({
+    tagName: 'toniq-dropdown',
+    props: {
+        options: [] as Readonly<ToniqDropdownOption[]>,
+        selected: undefined as undefined | Readonly<ToniqDropdownOption>,
+        dropdownOpen: false,
+    },
+    events: {
+        selectChange: defineElementEvent<ToniqDropdownOption>(),
+    },
+    styles: css`
+        :host {
+            ${toniqFontStyles.boldParagraphFont};
+        }
+
+        .dropdown {
+            width: 288px;
+            height: 48px;
+            position: relative;
+            border: ${removeNativeButtonStyles}
+            border-radius: 8px;
+        }
+
+        ${ToniqIcon} {
+            transition: ${interactionDuration} linear;
+        }
+
+        .dropdown.open ${ToniqIcon} {
+            transform: rotate(180deg);
+        }
+
+        .dropdown.open .select-options {
+            display: grid;
+        }
+
+        .dropdown.open .select {
+            border-bottom-left-radius: 0px;
+            border-bottom-right-radius: 0px;
+        }
+
+        .select {
+            display: flex;
+            justify-content: space-between;
+            border-radius: 8px;
+            cursor: pointer;
+            padding: 12px;
+            padding-left: 16px;
+            ${noUserSelect};
+            ${applyBackgroundAndForeground(toniqColors.accentSecondary)}
+        }
+
+        .select-options {
+            display: none;
+            position: absolute;
+            top: 100%;
+            left: 0;
+            right: 0;
+            z-index: 99;
+            border-radius: 8px;
+            ${applyBackgroundAndForeground(toniqColors.pagePrimary)}
+            filter: drop-shadow(0px 16px 16px rgba(39, 170, 243, 0.16));
```

Comment:
The color here looks like it should be some variation of `toniqColors.dropShadow.backgroundColor`?